docs/howto/krustlet-on-gke.md

```diff
@@ -0,0 +1,294 @@
+
+```shell
+$ IP=$(gcloud compute instances describe ${INSTANCE} \
+--project=${PROJECT} \
+--zone=${ZONE} \
+--format="value(networkInterfaces[0].networkIP)") && echo ${IP}
+```
+
+It should take less than 30-seconds to provision the VM.
+
+## Step 2: Create Certificate
+
+Krustlet requires a certificate for securing communication with the Kubernetes API. Because
+Kubernetes has its own certificates, we'll need to get a signed certificate from the Kubernetes API
+that we can use.
+
+In order for the Kubernetes cluster to resolve the name of the VM running the Krustlet, we'll include aliases for the VM's name in the certificate.
+
+First things first, let's create a certificate signing request (CSR):
+
+```shell
+$ ALIASES="DNS:${INSTANCE},DNS:${INSTANCE}.${ZONE},IP:${IP}"
+$ openssl req -new -sha256 -newkey rsa:2048 -keyout ./krustlet.key -out ./krustlet.csr -nodes -config <(
+cat <<-EOF
+[req]
+default_bits = 2048
+prompt = no
+default_md = sha256
+req_extensions = req_ext
+distinguished_name = dn
+[dn]
+O=.
+CN=${INSTANCE}
+[req_ext]
+subjectAltName = ${ALIASES}
+EOF
+)
+Generating a RSA private key
+....................+++++
+............................................+++++
+writing new private key to './krustlet.key'
+```
+
+This will create a CSR and a new key for the certificate. Now that it is created, we'll need to send the request to Kubernetes:
+
+```shell
+$ cat <<EOF | kubectl apply --filename=-
+apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+  name: krustlet
+spec:
+  request: $(cat krustlet.csr | base64 | tr -d '\n')
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+EOF
+certificatesigningrequest.certificates.k8s.io/krustlet created
+```
+
+You should then approve the request:
+
+```shell
+$ kubectl certificate approve krustlet
+certificatesigningrequest.certificates.k8s.io/krustlet approved
+```
+
+After approval, you can download the cert like so:
+
+```shell
+$ kubectl get csr krustlet --output=jsonpath='{.status.certificate}' \
+    | base64 --decode > krustlet.crt
+```
+
+Lastly, combine the key and the cert into a PFX bundle, choosing your own password instead of
+"password":
+
+```shell
+$ openssl pkcs12 -export -out krustlet.pfx -inkey krustlet.key -in krustlet.crt -password "pass:password"
+```
+
+## Step 3: Copy assets to VM
+
+The first thing we'll need to do is copy up the assets we generated in steps 1 and 2. Copy them to the VM by typing:
+
+```shell
+$ gcloud compute scp krustlet.pfx ${INSTANCE}: --project=${PROJECT} --zone=${ZONE}
+$ gcloud compute scp kubeconfig-sa ${INSTANCE}: --project=${PROJECT} --zone=${ZONE}
+```
+
+We can then SSH into the instance by typing:
+
+```shell
+$ gcloud compute ssh ${INSTANCE} --project=${PROJECT} --zone=${ZONE}
+```
+
+## Step 4: Install and configure Krustlet
```

Comment:
So the other cloud provider instructions have systemd units for their examples, whereas this is just running it in a terminal. Do you think it would be better to have those instructions here? You could probably mostly copy them from the AKS instructions

Reply:
Fixed: f3425d481fd6a7976e89a2d2ed405312e7c2be1d